src/main/java/me/rkfg/xmpp/bot/plugins/FaggotOfTheDayPlugin2.java

```diff
@@ -0,0 +1,198 @@
+        final Occupant occupant = Optional
+                .of(getMUCManager().listMUCOccupantsByJID(roomJid).get(faggotJid))
+                .orElse(faggot);
+
+        return INFO_FAGGOT_IS + occupant.getNick() + ".";
+    }
+
+    @Override
+    public List<String> getCommand() {
+        return ALL_COMMANDS;
+    }
+
+    @Override
+    public String getManual() {
+        final String commands = StringUtils.join(ALL_COMMANDS, "|");
+        final String sampleCommand = PREFIX + ALL_COMMANDS.get(0);
+
+        return
+                "узнать, кто сегодня Пидор дня.\n" +
+                "Формат: <" + commands + ">\n" +
+                "Пример: " + sampleCommand;
+    }
+
+    private void startListening() {
+        if (isListening) {
+            return;
+        }
+
+        isListening = true;
+
+        getMUCManager().listMUCs().forEach(muc -> {
+            muc.addMessageListener(message -> {
+               occupants.add(muc.getOccupant(message.getFrom()));
+            });
+
+            occupants.addAll(getAllOccupants(muc));
+        });
+    }
+
+    private void startTimer() {
+        if (timer != null) {
+            return;
+        }
+
+        timer = new Timer();
+        timer.scheduleAtFixedRate(new TimerTask() {
+
+            @Override
+            public void run() {
+                logger.info("🌚 A new day starts in the Empire 🌝");
+
+                occupants.removeIf(occupant -> {
+                    final String bareJid = XmppStringUtils.parseBareJid(occupant.getJid());
+                    return bareJid.equals(getBotJid());
+                });
+
+                if (occupants.isEmpty()) {
+                    logger.info("No contenders for Faggot of the Day today.");
+                    if (faggot != null) {
+                        logger.info("{} remains Faggot of the Day!", faggot.getNick());
+                    }
+                    return;
+                }
+
+                final String contenders = occupants.stream().map(Occupant::getNick)
+                            .collect(Collectors.joining(", "));
+                logger.info("Contenders for today’s Faggot of the Day title: {}", contenders);
+
+                final int i = random.nextInt(occupants.size());
+                faggot = occupants.stream().skip(i).findFirst().get();
+                logger.info("{} becomes Faggot of the Day!", faggot.getNick());
+
+                occupants.clear();
+            }
+
+        }, getFirstTime(), getPeriod());
+    }
+
+    private Date getFirstTime() {
+        final LocalTime midnight = LocalTime.MIDNIGHT;
+        final LocalDate today = LocalDate.now();
+        final LocalDateTime todayMidnight = LocalDateTime.of(today, midnight);
+        return Date.from(todayMidnight.atZone(ZoneId.systemDefault()).toInstant());
+    }
+
+    private long getPeriod() {
+        return TimeUnit.MILLISECONDS.convert(1, TimeUnit.DAYS);
```

Comment:
This code is shitty bad for me. Could not do both in the cruise.

Hint: use TimeUnit.DAYS.toMillis(1) and make this static final for better readability.